python/lib/adaguc/runAdaguc.py

```diff
@@ -34,6 +41,14 @@ def __init__(self):
         self.ADAGUC_TMP = os.getenv("ADAGUC_TMP", "/tmp")
         self.ADAGUC_FONT = os.getenv(
             "ADAGUC_FONT", self.ADAGUC_PATH + "/data/fonts/Roboto-Medium.ttf")
+        self.ADAGUC_REDIS = os.getenv("ADAGUC_REDIS", "")
+        if self.ADAGUC_REDIS.startswith("redis://") or self.ADAGUC_REDIS.startswith("rediss://"):
+            self.redis_url = self.ADAGUC_REDIS
+            self.use_cache = True
+            self.redis_pool = redis.ConnectionPool.from_url(self.ADAGUC_REDIS)
```

Comment:
Looking at it again, it does look like a fres runAdaguc instance is made for each request. So the pool should be come a class level variable. Sorry, my mistake.

Reply:
Converted into a class member